tests/core/full_node/test_mempool.py

```diff
@@ -276,253 +282,120 @@ async def test_invalid_block_index(self, two_nodes):
 
         for block in blocks:
             await full_node_1.full_node.respond_block(full_node_protocol.RespondBlock(block))
-        await time_out_assert(60, node_height_at_least, True, full_node_1, start_height + 3)
 
-        cvp = ConditionWithArgs(
-            ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE,
-            [uint64(start_height + 5).to_bytes(4, "big")],
-        )
-        dic = {ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE: [cvp]}
+        await time_out_assert(60, node_height_at_least, True, full_node_1, start_height + num_blocks)
 
-        spend_bundle1 = generate_test_spend_bundle(list(blocks[-1].get_included_reward_coins())[0], dic)
+        spend_bundle1 = generate_test_spend_bundle(
+            coin or list(blocks[-num_blocks + 2].get_included_reward_coins())[0], dic, uint64(fee)
```

Comment:
this `-num_blocks + 2` may look a bit questionable. The pattern we have in these tests is that almost all tests make 3 blocks and then looks at `-1`, i.e. the last one. There is *one* test that makes 4 blocks and looks at `-2`, so this formula works for the tests we have now.